src/coreclr/jit/codegenxarch.cpp

```diff
@@ -2646,8 +2632,7 @@ void CodeGen::genLclHeap(GenTree* tree)
         // adds to ESP).
 
         inst_RV(INS_NEG, regCnt, TYP_I_IMPL);
-        regNumber regTmp = tree->GetSingleTempReg();
```

Comment:
You removed the use of a temporary register here. That means we don't need to request that temporary register from LSRA. You need to figure out how to not request that temp for this specific case in `LinearScan::BuildLclHeap()`.